src/vs/workbench/parts/debug/electron-browser/debugService.ts

```diff
@@ -282,7 +284,15 @@ export class DebugService extends ee.EventEmitter implements debug.IDebugService
 
 		this.toDispose.push(this.session.addListener2(debug.SessionEvents.OUTPUT, (event: DebugProtocol.OutputEvent) => {
 			if (event.body && event.body.category === 'telemetry') {
-				this.telemetryService.publicLog(event.body.output, event.body.data);
+				const key = this.configurationManager.getAdapter().aiKey;
+				// only log telemetry events from debug adapter if the adapter provided the telemetry key
+				if (key) {
+					if (!this.telemetryAdapter) {
+						this.telemetryAdapter = new AIAdapter(key, this.session.getType());
```

Comment:
we need here to append common vscode properties such as machineId and sessionId, I will go ahead and merge this and then push a fix for that